qiskit_nature/properties/electronic_structure/integrals/electronic_integrals.py

```diff
@@ -0,0 +1,141 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""A base class for raw electronic integrals."""
+
+from abc import ABC, abstractmethod
+from typing import List, Optional, Tuple, Union
+
+import itertools
+
+import numpy as np
+
+from qiskit_nature.operators.second_quantization import FermionicOp
+
+from ..bases import ElectronicBasis, ElectronicBasisTransform
+
+
+class ElectronicIntegrals(ABC):
+    """A container for raw electronic integrals.
+
+    This class is a template for ``n``-body electronic integral containers.
+    It provides method stubs which must be completed in order to allow basis transformation between
+    different ``ElectronicBasis``. An extra method stub must be implemented to map into the special
+    ``ElectronicBasis.SO`` basis which is a required intermediate representation of the electronic
+    integrals during the process of mapping to a
+    ``qiskit_nature.operators.second_quantization.SecondQuantizedOp``.
+    """
+
+    def __init__(
+        self,
+        num_body_terms: int,
+        basis: ElectronicBasis,
+        matrices: Union[np.ndarray, Tuple[Optional[np.ndarray], ...]],
+    ) -> None:
+        """
+        Args:
+            num_body_terms: ``n``, as in the ``n-body`` terms stored in these integrals.
+            basis: the basis which these integrals are stored in. If this is initialized with
+                ``ElectronicBasis.SO``, these integrals will be used *ad verbatim* during the
+                mapping to a ``SecondQuantizedOp``.
+            matrices: the matrices (one or many) storing the actual electronic integrals. If this is
+                a single matrix, ``basis`` must be set to ``ElectronicBasis.SO``. Refer to the
+                documentation of the specific ``n-body`` integral types for the requirements in case
+                of multiple matrices.
+        """
+        self._basis = basis
+        assert num_body_terms >= 1
```

Comment:
I think we should be raising ValueError here or TypeError here as these are public APIs and users can pass whatever. Asserts are more for internal state checks to assure what we expect there meets whatever the condition is for debugging etc